### ISSUE TITLE
Rhino gives usage warning when certain Java objects are placed in variable maps

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#2964]
This doesn't appear to actually affect message processing, Rhino just outputs a "Missed Context.javaToJS() conversion" warning. And it only appears to happen for certain Java objects. For example this will produce the warning:
\\
\\
`$co('globalChannelMap', globalChannelMap);`
\\
But this will not:
\\
\\
`$co('a map', new java.util.HashMap());`

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3046
Reporter: narupley
Created: 2013-11-07T08:51:13.000-0800

**Comment by rbeckman-nextgen** [comment]
Fixed in revision 7017. The issue was that the "value === undefined" check in the $ functions were actually passing the Java object reference into the Rhino equality operator, and for certain classes Rhino doesn't know what to do and logs the warning (the condition still evaluates correctly as false though). To avoid that, we changed the check back to "arguments.length == 1".

Imported Comment. Original Details:
Author: narupley
Created: 2013-11-08T11:00:25.000-0800
